src/app/MessageDef/SubscribeRequest.h

```diff
@@ -0,0 +1,220 @@
+     *
+     *  @return #CHIP_NO_ERROR on success
+     *          #CHIP_END_OF_TLV if there is no such element
+     */
+    CHIP_ERROR GetAttributeDataVersionList(AttributeDataVersionList::Parser * const apAttributeDataVersionList) const;
+
+    /**
+     *  @brief Get Event Number. Next() must be called before accessing them.
+     *
+     *  @return #CHIP_NO_ERROR on success
+     *          #CHIP_END_OF_TLV if there is no such element
+     */
+    CHIP_ERROR GetEventNumber(uint64_t * const apEventNumber) const;
+
+    /**
+     *  @brief Get Min Interval. Next() must be called before accessing them.
+     *
+     *  @return #CHIP_NO_ERROR on success
+     *          #CHIP_END_OF_TLV if there is no such element
+     */
+    CHIP_ERROR GetMinInterval(uint16_t * const apMinInterval) const;
+
+    /**
+     *  @brief Get Max Interval. Next() must be called before accessing them.
+     *  @return #CHIP_NO_ERROR on success
+     *          #CHIP_END_OF_TLV if there is no such element
+     */
+    CHIP_ERROR GetMaxInterval(uint16_t * const apMaxInterval) const;
+
+    /**
+     *  @brief Check if subscription is kept. Next() must be called before accessing them.
+     *  @return #CHIP_NO_ERROR on success
+     *          #CHIP_END_OF_TLV if there is no such element
+     */
+    CHIP_ERROR GetKeepExistingSubscriptions(bool * const apKeepExistingSubscription) const;
+
+    /**
+     *  @brief Check if subscription is kept. Next() must be called before accessing them.
+     *  @return #CHIP_NO_ERROR on success
+     *          #CHIP_END_OF_TLV if there is no such element
+     */
+    CHIP_ERROR GetIsProxy(bool * const apIsProxy) const;
+};
+
+class Builder : public chip::app::Builder
+{
+public:
+    /**
+     *  @brief Initialize a SubscribeRequest::Builder for writing into a TLV stream
+     *
+     *  @param [in] apWriter    A pointer to TLVWriter
+     *
+     *  @return #CHIP_NO_ERROR on success
+     */
+    CHIP_ERROR Init(chip::TLV::TLVWriter * const apWriter);
+
+    /**
+     *  @brief Initialize a AttributePathList::Builder for writing into the TLV stream
+     *
+     *  @return A reference to AttributePathList::Builder
+     */
+    AttributePathList::Builder & CreateAttributePathListBuilder();
+
+    /**
+     *  @brief Initialize a EventPathList::Builder for writing into the TLV stream
+     *
+     *  @return A reference to EventPathList::Builder
+     */
+    EventPathList::Builder & CreateEventPathListBuilder();
+
+    /**
+     *  @brief Initialize a AttributeDataVersionList::Builder for writing into the TLV stream
+     *
+     *  @return A reference to AttributeDataVersionList::Builder
+     */
+    AttributeDataVersionList::Builder & CreateAttributeDataVersionListBuilder();
+
+    /**
+     *  @brief An initiator can optionally specify an EventNumber it has already to limit the
+     *  set of retrieved events on the server for optimization purposes.
+     *  @param [in] aEventNumber The event number
```

Comment:
remove `@param` item - it is obvious that EventNumber/aEventNumber refers to an event number.

Reply:
Done, thanks